comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -30,16 +39,22 @@ const (
 )
 
 type collector struct {
-	id                  string
-	store               workloadmeta.Component
-	catalog             workloadmeta.AgentType
-	metaV1              v1.Client
-	metaV3or4           func(metaURI, metaVersion string) v3or4.Client
-	clusterName         string
-	hasResourceTags     bool
-	collectResourceTags bool
-	resourceTags        map[string]resourceTags
-	seen                map[workloadmeta.EntityID]struct{}
+	id                      string
+	store                   workloadmeta.Component
+	catalog                 workloadmeta.AgentType
+	metaV1                  v1.Client
+	metaV3or4               func(metaURI, metaVersion string) v3or4.Client
+	clusterName             string
+	hasResourceTags         bool
+	collectResourceTags     bool
+	resourceTags            map[string]resourceTags
+	seen                    map[workloadmeta.EntityID]struct{}
+	v4TaskEnabled           bool
+	v4TaskCache             *cache.Cache
+	v4TaskRefreshInterval   time.Duration
+	v4TaskQueue             []string
```

Comment:
I think here you can use a a queue like `workqueue.RateLimitingInterface` to simplify the code and reuse exiting queue mechanism.

Reply:
That's a good point. I've added a worker which is used by the collector to get tasks from endpoint
https://github.com/DataDog/datadog-agent/blob/66b205de2f9722a8c203d30fae8c5a1d10514779/comp/core/workloadmeta/collectors/internal/ecs/worker.go#L23-L27